perfkitbenchmarker/flag_util.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions for working with user-supplied flags."""
+
+import logging
+import re
+
+INTEGER_GROUP_REGEXP = re.compile(r'(\d+)(-(\d+))?$')
+
+
+def ParseIntegerList(inp, on_nonincreasing=None):
```

Comment:
After looking at it for a bit longer, I can write a DEFINE_-style function like the ones in gflags, but I don't think I can implement a serializer in the way I'd want to. I think the right way to do it is to parse the string input into an intermediate list that stores either single integers or tuples giving the bounds on ranges. You can easily go from this to either an iterable over all of the integers, equivalent to the list the ParseIntegerList function returns, or back to the original string representation for your serializer.

When PKB first starts up, it prints out all of its command-line flags using serializers (https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/pkb.py#L326). This would break that logging, in a way that having people use DEFINE_string and register validators would not.

I could make a function called DEFINE_integerlist that looks like a flags function but doesn't really work the same way, but I'm afraid that would break things that depend on the flags interface. However, I don't see any other options. What do you think?